### ISSUE TITLE
Tree shaking doesn't work if a class has any field

### ISSUE THREAD

**Issue by AlttiRi** [rollup/rollup#3564]
Tree shaking doesn't work if a class has any field (static or not).
---

### [Here is the example, that works OK (the classes have only a method):](https://rollupjs.org/repl/?version=2.10.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QkJhciU3RCUyMGZyb20lMjAlNUMlMjIuJTJGbW9kdWxlLmpzJTVDJTIyJTNCJTVDbiU1Q25jb25zdCUyMGJhciUyMCUzRCUyMG5ldyUyMEJhcigpJTNCJTVDbmNvbnNvbGUubG9nKGJhci5nZXREYXRhKCkpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybW9kdWxlLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGNsYXNzJTIwRm9vJTIwJTdCJTVDbiUyMCUyMGdldERhdGEoKSUyMCU3QiUyMHJldHVybiUyMCU1QyUyMkZPTyU1QyUyMiUzQiUyMCU3RCU1Q24lN0QlNUNuZXhwb3J0JTIwY2xhc3MlMjBCYXIlMjAlN0IlNUNuJTIwJTIwZ2V0RGF0YSgpJTIwJTdCJTIwcmV0dXJuJTIwJTVDJTIyQkFSJTVDJTIyJTNCJTIwJTdEJTVDbiU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)

_module.js_
```js
export class Foo {
    getData() { return "FOO"; }
}
export class Bar {
    getData() { return "BAR"; }
}
```
_main.js_
```js
import {Bar} from "./module.js";

const bar = new Bar();
console.log(bar.getData());
```

I run:
```js
const inputOptions = {input: "./main.js"};
const outputOptions = {file: "./dist/bundle.js"};
rollup.rollup(inputOptions).then(bundle => bundle.write(outputOptions));
```
And I receive the correct result:
_bundle.js_
```js
class Bar {
    getData() { return "BAR"; }
}

const bar = new Bar();
console.log(bar.getData());
```
---

### [**But if I add a field _(static or not)_ to the unused class `Foo`:**](https://rollupjs.org/repl/?version=2.10.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QkJhciU3RCUyMGZyb20lMjAlNUMlMjIuJTJGbW9kdWxlLmpzJTVDJTIyJTNCJTVDbiU1Q25jb25zdCUyMGJhciUyMCUzRCUyMG5ldyUyMEJhcigpJTNCJTVDbmNvbnNvbGUubG9nKGJhci5nZXREYXRhKCkpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybW9kdWxlLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGNsYXNzJTIwRm9vJTIwJTdCJTVDbiU1Q3Rmb28lM0IlNUNuJTIwJTIwZ2V0RGF0YSgpJTIwJTdCJTIwcmV0dXJuJTIwJTVDJTIyRk9PJTVDJTIyJTNCJTIwJTdEJTVDbiU3RCU1Q25leHBvcnQlMjBjbGFzcyUyMEJhciUyMCU3QiU1Q24lMjAlMjBnZXREYXRhKCklMjAlN0IlMjByZXR1cm4lMjAlNUMlMjJCQVIlNUMlMjIlM0IlMjAlN0QlNUNuJTdEJTIyJTJDJTIyaXNFbnRyeSUyMiUzQWZhbHNlJTdEJTVEJTJDJTIyb3B0aW9ucyUyMiUzQSU3QiUyMmZvcm1hdCUyMiUzQSUyMmVzJTIyJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTdEJTJDJTIyZXhhbXBsZSUyMiUzQW51bGwlN0Q=)

_module.js_
```js
export class Foo {
    foo;
    getData() { return "FOO"; }
}
export class Bar {
    getData() { return "BAR"; }
}
```
**I get the bundle with the unused class `Foo`:**
_bundle.js_
```js
class Foo {
    foo;
    getData() { return "FOO"; }
}
class Bar {
    getData() { return "BAR"; }
}

const bar$1 = new Bar();
console.log(bar$1.getData());
```
**The expected result should be like this _(with no `Foo` class)_:** 
_bundle.js_
```js
class Bar {
    getData() { return "BAR"; }
}

const bar = new Bar();
console.log(bar.getData());
```

---
Rollup.js 2.10.2

**Comment by AlttiRi** [comment]
[Should note that it works well with **private class fields** (whose name starts with `#`)](https://rollupjs.org/repl/?version=2.10.2&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmltcG9ydCUyMCU3QkJhciU3RCUyMGZyb20lMjAlNUMlMjIuJTJGbW9kdWxlLmpzJTVDJTIyJTNCJTVDbiU1Q25jb25zdCUyMGJhciUyMCUzRCUyMG5ldyUyMEJhcigpJTNCJTVDbmNvbnNvbGUubG9nKGJhci5nZXREYXRhKCkpJTNCJTIyJTJDJTIyaXNFbnRyeSUyMiUzQXRydWUlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIybW9kdWxlLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMGNsYXNzJTIwRm9vJTIwJTdCJTVDbiUyMCUyMCUyMCUyMCUyM2ZvbyUzQiU1Q24lMjAlMjAlMjAlMjBzdGF0aWMlMjAlMjNzRm9vJTNCJTVDbiUyMCUyMCUyMCUyMGdldERhdGEoKSUyMCU3QiUyMHJldHVybiUyMCU1QyUyMkZPTyU1QyUyMiUzQiUyMCU3RCU1Q24lN0QlNUNuZXhwb3J0JTIwY2xhc3MlMjBCYXIlMjAlN0IlNUNuJTIwJTIwJTIwJTIwJTIzYmFyJTNCJTVDbiUyMCUyMCUyMCUyMHN0YXRpYyUyMCUyM3NCYXIlM0IlNUNuJTIwJTIwJTIwJTIwZ2V0RGF0YSgpJTIwJTdCJTIwcmV0dXJuJTIwJTVDJTIyQkFSJTVDJTIyJTNCJTIwJTdEJTVDbiU3RCUyMiUyQyUyMmlzRW50cnklMjIlM0FmYWxzZSU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMm5hbWUlMjIlM0ElMjJteUJ1bmRsZSUyMiUyQyUyMmFtZCUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyJTIyJTdEJTJDJTIyZ2xvYmFscyUyMiUzQSU3QiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE)

_module.js_
```js
export class Foo {
    #foo;
    static #sFoo;
    getData() { return "FOO"; }
}
export class Bar {
    #bar;
    static #sBar;
    getData() { return "BAR"; }
}
```
I get the correct result:

_bundle.js_
```js
class Bar {
    #bar;
    static #sBar;
    getData() { return "BAR"; }
}

const bar = new Bar();
console.log(bar.getData());
```

**Comment by AlttiRi** [comment]
Also can note that with Rollup version below 2.4.0 (https://github.com/rollup/rollup/pull/3488) no one of my examples does not work.

**Comment by lukastaegert** [comment]
Thanks for spotting, fix at #3569
